### PR TITLE
Lint views/courses and fix up run_moss UI

### PR DESCRIPTION
**Pull request by 20wildmanj**
### Summary generated by Reviewpad on 12 Sep 23 19:33 UTC
This pull request includes changes to multiple files. Here is a summary of the modifications:

1. In `reload.html.erb`, changes are made to the options section. Two list items are replaced with multiline list items. The first list item now includes a link to reload the configuration file with an updated data method. The second list item remains the same.

2. The file `app/views/courses/download_roster.html.erb` has been deleted. It used to contain a loop that prints out various attributes of users in a specific format. The loop iterates over `@users`, and each line printed includes the semester, user details, school, major, year, grade policy, lecture, and section.

3. In `report_bug.html.erb`, the indentation of the `text_area_tag`, options for the `text_area_tag`, and options for the `submit_tag` have been reformatted to improve readability and maintainability of the code.

4. The file `app/views/courses/_updateRosterPreview.html.erb` has undergone several changes, including replacing the closing tag of the checkbox input, changing the syntax of the class attribute in the submit buttons, changing the value of the `locals` hash key `sort_status`, and removing an extra new line at the end of the file.

5. Changes in `courses/_usersTable.html.erb` involve removing whitespace and simplifying the conditional statement for displaying user roles.

6. In `app/views/courses/_uploadError.html.erb`, changes include removing the trailing space in the `<br />` tag and changing it to `<br>`, as well as removing the trailing space in the `<table>` tag.

7. The file `app/views/courses/_uploadTable.html.erb` has changes related to formatting and styling, including removing unnecessary spaces in the code, updating attribute styles, and adding a newline at the end of the file.

8. In `index.html.erb` in the `app/views/courses` directory, a link is modified to include a section parameter, and an exception handling block is added for rendering the partial `home/topannounce`.

9. Various changes have been made to improve code formatting and maintainability without affecting the logic or functionality in the modified files.

10. Changes in `app/views/courses/email.html.erb` include updating variables, wrapping them in `<font>` tags with different colors, aligning and indenting the section select options, updating the form tag with the symbol syntax, removing quotes for the table class attribute, and using hash syntax for options and attributes.

11. Changes in `upload_roster.html.erb` involve fixing spacing issues, adjusting indentation, updating routes and form syntax to follow best practices, and adding labels and proper alignment.

12. `app/views/components/_dropdown_icon.html.erb` is a new file that adds icons and JavaScript code for toggling visibility of certain elements based on user interaction in the UI.

13. Changes in `users.html.erb` include updating routes, adjusting indentation, and adding attributes for better UX.

14. The file diff includes several changes related to styling updates for the Moss page, including adding new classes, changing backgrounds, and applying padding, margin, and border-radius.

15. Changes in the file involve updating HTML tags and attributes, adjusting indentation, adding help text, and fixing class attributes.

16. In `manage.html.erb`, several instances of the `link_to` method have been replaced with updated routing helpers for course and management actions.

17. In `run_moss.html.erb`, changes include updating the heading tag, adding a check for job success, displaying messages and results, and including a heading with command string and result output.

18. The file diff of `app/views/courses/_courseFields.html.erb` includes changes to indentation, formatting, and help text attributes for various fields.

19. The file `makeDlist.html.erb` has been deleted in this diff. It displayed error or success messages.

Please let me know if there's anything else you need assistance with.

## Description

- lint `views/courses` using `bundle exec erblint app/views/courses/*(file).html.erb -a`
- Fix up `moss.html.erb` and `run_moss.html.erb` to be more bearable on the eyes by modifying styling, adding `components/_dropdown_icon.html.erb` for dropdown icon html and jquery
- Turn `upload_roster.html.erb` form into `FormBuilderWithDateTimeInput` to have styling on upload file button
- Remove unused files

#### Before

Screenshot 2023-08-30 at 10 53 23 PM
Screenshot 2023-08-30 at 10 53 27 PM

- Ow my eyes, also no dropdown arrow

After:

Screenshot 2023-08-30 at 10 52 51 PM
Screenshot 2023-08-30 at 10 52 56 PM

- acceptable

#### Before

Screenshot 2023-08-30 at 11 31 24 PM
Screenshot 2023-08-30 at 11 32 01 PM

#### After

Screenshot 2023-08-30 at 11 10 38 PM
Screenshot 2023-08-30 at 11 30 58 PM

## Motivation and Context

- Summer linting

## How Has This Been Tested?

- check each page / erb file renders correctly

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run rubocop for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting

### CODE REVIEW

File: app/views/components/_dropdown_icon.html.erb
Comment (damianhxy) [inline app/views/components/_dropdown_icon.html.erb:1]: I realize that this is the opposite behavior of the global annotations dropdown. This seems to be the standard, however, so we might want to change the annotations dropdown icon.

File: app/views/courses/moss.html.erb [hunk 1]
```diff
@@ -3,164 +3,183 @@
...
+  <% end %>
 
-		</script>
-	<% end %>
+  <h4>Step 1:</h4>
+  <p>Check the box for each assessment you want to send to Moss for cheat checking.</p>
 
-	<h4>Step 1:</h4>
-	<p>Check the box for each assessment you want to send to Moss for cheat checking.</p>
+  <%= form_tag([:run_moss, @course], multipart: true) do %>
```
Comment (damianhxy) [inline app/views/courses/moss.html.erb:29]: Can this be changed to `run_moss_course_path`?

File: app/views/courses/upload_roster.html.erb [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
-    <span>Mark students not included in this roster as 'Dropped'?</span> <br />
-    <font class="smallText">Dropped students can view grades but cannot submit
-      assignments or download new assignments</font>
-</label>
-<p></p>
-<%= f.submit 'Upload!' , {:class=>"btn primary"} %>
-<% end %>
+  <p>
+    The CMU Roster file should include a header.  Editing accounts will occur as a
+    <a href="http://en.wikipedia.org/wiki/Database_transaction#Transactional_databases">transaction</a>,
```
Comment (damianhxy) [inline app/views/courses/upload_roster.html.erb:18]: nit: change this to a https link

File: app/views/courses/upload_roster.html.erb [hunk 3]
```diff
@@ -1,35 +1,40 @@
...
+    if there is an error we will cancel all changes and allow you to correct the error and re-submit.
+  </p>
+
+  <%= form_for :upload, url: { action: "upload_roster" },
```
Comment (damianhxy) [inline app/views/courses/upload_roster.html.erb:22]: nit: change this to `upload_roster_course_path`

File: app/views/courses/users.html.erb [hunk 4]
```diff
@@ -15,8 +15,8 @@
     <%= link_to raw('<span class="btn">Add User to Course</span>'), [:new, @course, :course_user_datum] %>
     <!-- Add User Dropdown Trigger -->
     <a id="add-users" class='dropdown-trigger btn' href='#' data-target='add-users-dropdown'>ADD USERS FROM EMAILS</a>
-    <%= link_to raw('<span class="btn">Import Roster/Users</span>'), {:action=>"upload_roster"} %>
-    <%= link_to raw('<span class="btn">Export Roster</span>'), {:action=>"download_roster"} %>
+    <%= link_to raw('<span class="btn">Import Roster/Users</span>'), { action: "upload_roster" } %>
+    <%= link_to raw('<span class="btn">Export Roster</span>'), { action: "download_roster" } %>
```
Comment (damianhxy) [inline app/views/courses/users.html.erb:19]: nit: update to use `download_roster_course_path` / `upload_roster_course_path`

File: app/views/courses/users.html.erb [hunk 5]
```diff
@@ -32,17 +32,23 @@
...
       </div>
       <div class="modal-footer">
         <span style="float: left;">
-          <%= f.submit "Save" , {class: "btn-flat"} %>
+          <%= f.submit "Save", { class: "btn-flat" } %>
           <a href="#!" class="modal-close btn-flat">Cancel</a>
         </span>
-        <%= link_to raw('<span class="btn-flat">Unlink Course</span>'), {action: "unlink_course"}, data: { confirm: "Are you sure you want to unlink the course?" }, method: :post %>
+        <%= link_to raw('<span class="btn-flat">Unlink Course</span>'),
+                    { action: "unlink_course" },
```
Comment (damianhxy) [inline app/views/courses/users.html.erb:49]: nit: `unlink_course_course_path`

**Review by damianhxy — APPROVED** [review]
- Checked that links on manage course page work
- Checked that MOSS works (success and failure)
- Checked that roster import works

LGTM (but it would be great to convert some remaining actions to new-style path helpers)

File: app/views/components/_dropdown_icon.html.erb
Comment (20wildmanj) [reply app/views/components/_dropdown_icon.html.erb:1]: Which icon are you referring to? The dropdowns on annotations seem fine to me:

Comment (damianhxy) [reply app/views/components/_dropdown_icon.html.erb:1]: Nevermind I was tripping

**Pull request merged**